### PR TITLE
doc：Add new loaderUpdate.md

### PR DESCRIPTION
**Pull request by Maledong**
1. This will fix the issue eggjs/egg-core#195 and gives a good answer to #3362. We should notify users how to upgrade their lifecyle events quickly.

2. Fix some '##title', we should directly use 'title:...' instead.

3. Due to the conflict of typescript of nodejs in the global/local, we've forcely defined the type path.

---
- [X] `npm test` passes
- [ ] tests and/or benchmarks are included
- [X] documentation is changed or added
- [X] commit message follows commit guidelines

### CODE REVIEW

File: docs/source/zh-cn/advanced/loaderUpdate.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+module.exports = AppBootHook;
+```
+## ready 函数替代
+
+同样地，我们之前在 `app.ready` 中处理我们的逻辑：
+
+```js
+module.exports = app => {
+    app.beforeStart(async () => {
```
Comment (dead-horse) [inline docs/source/zh-cn/advanced/loaderUpdate.md]:
```js
app.ready(() => {
  // 
});
```

File: docs/source/zh-cn/advanced/loaderUpdate.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+   async beforeClose() {
+    // 请将您的 app.beforeClose 中的代码置于此处。
+  }
+}
+```
+## 其它说明
+
+本教程只是一对一地讲了替换方法，便于开发者们快速上手进行替换；若想要具体了解整个 Loader 原理以及生命周期的完整函数版本，请参考[加载器](./loader.md)这篇文章。
```
Comment (dead-horse) [inline docs/source/zh-cn/advanced/loaderUpdate.md]: 请参考[加载器](./loader.md)和[启动自定义](app-start.md)。

File: docs/source/_data/guide_toc.yml [hunk 3]
```diff
@@ -39,6 +39,7 @@ Tutorials:
   TypeScript: /tutorials/typescript.html
 Advanced:
   Loader: /advanced/loader.html
+  LoaderUpdate: /advanced/loaderUpdate.html
```
Comment (dead-horse) [inline docs/source/_data/guide_toc.yml]: 建议这篇文档不加到目录中，通过一个外链放在 app-start 这个章节里面即可。

Comment (Maledong) [reply docs/source/_data/guide_toc.yml]: OK，好的。

**Comment by Maledong** [comment]
@dead-horse：完成了，谢谢。
请注意，最近Egg的test运行很多通不过。

**Comment by popomore** [comment]
@whxaxes 是 ts. 挂了

**Comment by whxaxes** [comment]
@popomore  我今天处理一下

**Comment by Maledong** [comment]
@whxaxes：谢谢你。

**Pull request merged**